src/utils/filterResults.ts

```diff
@@ -163,6 +187,24 @@ export default function filterResults (unsortedPropertyResults: UnprocessedResul
                         if (ownerData.length > 0) {
                             property.owner_details = ownerData
                         }
+                        let nameOnTitlePropertyData = getNameonTitlePropertyData(property)
+                        property.NameOnTitlePropertyDetails = nameOnTitlePropertyData
+                        if (nameOnTitlePropertyData.length > 2) {
+                            for (let i = 0; i < nameOnTitlePropertyData.length; ++i)
+                                for (let j = 0; j < nameOnTitlePropertyData.length; ++j)
+                                    if (i !== j && nameOnTitlePropertyData[i].Name_On_Title.id === nameOnTitlePropertyData[j].Name_On_Title.id)
+                                    nameOnTitlePropertyData.splice(j, 1); 
+                            property.NameOnTitlePropertyDetails = nameOnTitlePropertyData
```

Comment:
use Array.find instead of nested for loops

Reply:
use to remove duplicates